### DISCUSSION TITLE
Unexpected Changes in Minification after Upgrading to 13.5

### DISCUSSION THREAD

**Discussion by banuni** [vercel/next.js#58182] [Help] [answered]
### Summary

Hey, after upgrading from 13.4 to 13.5 I'm encountering an issue in minifying:
Sadly, my project uses `FooClass.prototype.constructor.name` in key places, and now it seems to be broken
what was once received as `[class FooClass extends BaseClass]` now resolves to `[class n extends a]`
is there a new optimization I'm supposed to be able to opt-out of?``

### Additional information
* The classes reside in an internal package in the same monorepo.
* Just to be clear - this happens on a `next build` build command, `next dev` works as usual.

**Comment by icyJoseph** [comment]
Try turning off `swcMinify` in your `next.config.js`.

We can use the results of that to move forward.

**Reply by banuni** [reply]
thanks. this didn't really help :(

maybe it'll be good to mention the classes are in separate, internal packages that reside in the same repository.

**Reply by icyJoseph** [reply]
Aha, but then it has to be in the webpack minification step 🤔 but why would that change...

**Reply by banuni** [reply]
I've removed all of my webpack modifications to the config and still get the same results.

**Reply by icyJoseph** [reply]
Yea, but like, I don't think it comes from your side. We'd need to look at the diff between those releases.

**Reply by banuni** [reply]
I tried, but it's a mere 1700 files change =[
[here](https://github.com/vercel/next.js/compare/v13.4.19...v13.5.1)

**Reply by icyJoseph** [reply]
Uff, and it is a lot of canaries in between too... but I guess, you can binary search it, take a canary in the middle, install it and test, and then go forward or backwards

**Reply by banuni** [reply]
Yup, i've been doing it since the last message, I'll find something soon, I hop

**Reply by banuni** [reply]
found it! it's between canary 18 and 19:
https://github.com/vercel/next.js/compare/v13.4.20-canary.18...v13.4.20-canary.19
if you have any suggestion or hint I'd really appreciate it!

**Reply by icyJoseph** [reply]
Try with this then:

https://github.com/vercel/next.js/commit/30da48fcd28b1fc07dcfa06d89b28bec95897d10

Opting out of server side minification:

> you can opt out via specifying `experimental.serverMinification: false` in `next.config.js`

**Reply by banuni** [reply]
It's this commit!!! fixed it!
https://github.com/vercel/next.js/commit/30da48fcd28b1fc07dcfa06d89b28bec95897d10

**Reply by banuni** [reply]
awesome! do you think we need to document this somewhere?

**Comment by banuni** [comment, marked as answer]
After a tedious binary search in canary versions between minor versions (41 of those!)
the solution was to opt-out of the now default experimental feature of serverMinification:
```ts
//next.config.js
module.exports = {
    experimental: {
        serverMinification: false,
        ....
    },
    ....
}
```
the problem occurred because of this [commit](https://github.com/vercel/next.js/commit/30da48fcd28b1fc07dcfa06d89b28bec95897d10)
also many thanks to @icyJoseph who was there for me when I needed <3
